hack/cmd/deps/deps.go

```diff
@@ -450,3 +463,294 @@ func loadGoToolJSON[Item any](args ...string) ([]Item, error) {
+		content = strings.ReplaceAll(content, "${WEEKLY_RUN}", weeklyRun)
+		content = strings.ReplaceAll(content, "${DEFAULT_BRANCH}", defaultBranchStanza)
+		_, _ = data.WriteString(content)
+	}
+
+	return os.WriteFile(file, data.Bytes(), 0644)
+}
+
+func indentBlocks(blocks []templateData) []templateData {
+	for i, block := range blocks {
+		lines := strings.Split(block.content, "\n")
+		indented := ""
+		for _, line := range lines {
+			if line == "" {
+				continue
+			}
+			indented += "  " + line + "\n"
+		}
+		blocks[i].content = indented
+	}
+	return blocks
+}
+
+func readTemplates(templatesDir string) ([]templateData, error) {
+	templateFiles, err := os.ReadDir(templatesDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read templates directory: %w", err)
+	}
+	var templates []templateData
+	for _, t := range templateFiles {
+		if t.IsDir() {
+			continue
+		}
+		if !strings.HasSuffix(t.Name(), ".yml") {
+			continue
+		}
+		origPath := filepath.Join(templatesDir, t.Name())
+		contentBytes, err := os.ReadFile(origPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read template file %s: %v", origPath, err)
+		}
+		content := string(contentBytes)
+		// Clean up extra space at end of file; but ensure there's one newline.
+		content = strings.TrimRight(content, "\n ")
+		content += "\n"
+		templates = append(templates, templateData{
+			originalPath: origPath,
+			filename:     t.Name(),
+			content:      content,
+		})
+	}
+	return templates, err
+}
+
+func mustReadFile(path string) []byte {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		logrus.Fatalf("Failed to read %s: %v", path, err)
+	}
+	return b
+}
+
+func extractChangeInPlaceholders(templates []templateData) set.Set[string] {
+	pattern := regexp.MustCompile(`\$\{CHANGE_IN\(([^)]+)\)}`)
+	out := set.New[string]()
+	for _, t := range templates {
+		matches := pattern.FindAllStringSubmatch(t.content, -1)
+		for _, m := range matches {
+			out.Add(m[1])
+		}
+	}
+	return out
+}
+
+func validateChangeInClauses(semaphoreDir string) error {
+	root := filepath.Join(semaphoreDir, "semaphore.yml.d")
+	var offending []string
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() || !strings.HasSuffix(path, ".yml") {
+			return nil
+		}
+		data := mustReadFile(path)
+		scanner := bufio.NewScanner(bytes.NewReader(data))
+		for scanner.Scan() {
+			l := scanner.Text()
+			if strings.Contains(l, "change_in(") && !strings.Contains(l, "pipeline_file:") {
+				offending = append(offending, fmt.Sprintf("%s: %s", path, l))
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	if len(offending) > 0 {
+		return fmt.Errorf("All change_in clauses must include pipeline_file: 'ignore' (or 'track'). Offending lines:\n%s", strings.Join(offending, "\n"))
+	}
+	return nil
+}
+
+func calculateBranchStanza(semaphoreDir string) string {
```

Comment:
It looks like this will work correctly and give "" for OSS.  But were you intending to rely on that, or did you mean to code another OSS/Enterprise switch somewhere?

Reply:
Have changed to use a regex that should detect OS/ent versions.